### PR TITLE
Creating new rule for SecretLint for using regular expressions

### PR DESCRIPTION
**Pull request by PseudoCoding**
Creating @secretlint/secretlint-rule-pattern for using regular expressions like below 

```json
{
  "rules": [
    {
      "id": "@secretlint/secretlint-rule-pattern",
      "options": {
        "patterns": [
          {
            "name": "password=",
            "pattern": "/password\\s*=\\s*(?<password>[\\w\\d!@#$%^&(){}\\[\\]:\";'<>,.?\/~`_+-=|]{1,256})\\b.*\/gi"
          },
          {
            "name": "client_secret=",
            "pattern": "/client_secret\\s*=\\s*(?<password>[\\w\\d!@#$%^&(){}\\[\\]:\";'<>,.?\/~`_+-=|]{1,256})\\b.*\/gi"
          }
        ]
      }
    }
  ]
}
```

Let me know what feedback you have. (Issue #130)

### CODE REVIEW

**Comment by azu** [comment]
I'm ok with `secretlint-rule-pattern`.

File: packages/@secretlint/secretlint-rule-pattern0/package-lock.json [hunk 1]
```diff
@@ -0,0 +1,1631 @@
+{
```
Comment (azu) [inline packages/@secretlint/secretlint-rule-pattern0/package-lock.json:1]: This repository use [Yarn v1](https://classic.yarnpkg.com/).

https://github.com/secretlint/secretlint/blob/master/CONTRIBUTING.md

`package-lock.json` will not needed.

File: packages/@secretlint/secretlint-rule-pattern0/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    for (const p of options.patterns) {
+      /**
+      * The next several lines were to allow the configuration to be the full regex string
+      * and to break it down to what the RegExp class expects
+      **/
+      const lastIndex = p.pattern.lastIndexOf('/')
+      const regexText = p.pattern.startsWith('/') ? p.pattern.substr(1, lastIndex - 1) : p.pattern.substr(0, lastIndex - 1)
+      const flags = p.pattern.endsWith('/') ? 'g' : p.pattern.substr(lastIndex + 1)
+      const regex = new RegExp(regexText, flags)
+      const results = source.content.matchAll(regex);
```
Comment (azu) [inline packages/@secretlint/secretlint-rule-pattern0/src/index.ts:47]: Maybe, It is just replaced with `matchPatterns(source.content, options.patterns);`
regexp-string-matcher has same feature.

https://github.com/textlint/regexp-string-matcher

**Comment by PseudoCoding** [comment]
Thanks for the feedback. Should be resolved now.

**Comment by azu** [comment]
Thanks! I will publish new version in tommorow.

**Pull request merged**